Automattic/jetpack
# Enable accessibility rules for eslint

This PR adds the accessibility configuration added to eslint-config-calypso in Automattic/eslint-config-wpcalypso#4, and fixes the warnings it generates.

> This PR enables the [jsx-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y) rules for ESLint. This will let us quickly check for a handful of accessibility issues common in javascript apps. It won't get us to 100% accessible, but it's a good step for awareness of these issues.

You can see [the supported rules](https://github.com/evcohen/eslint-plugin-jsx-a11y#supported-rules) for what this checks. Some of these have been disabled, you can see the comments in [`react-a11y/index.js`](https://github.com/Automattic/eslint-config-wpcalypso/blob/add/jsx-a11y-rules/react-a11y/index.js).

**To test:**

1. Run `yarn distclean && yarn` to update your packages
2. Run `npm run lint` to run the checks
3. This should only return warnings, no errors

**Note**: ~~This can't really be merged until the eslint-config-wpcalypso PR is merged & deployed, but by linking you can test & give feedback on the rules~~  The rules have been merged and `eslint-config-wpcalypso` has been updated, so this PR is OK to merge now 😄
